### PR TITLE
feat: use materialized view for usage explorer API

### PR DESCRIPTION
**Pull request by makeavish**
Otel collector PR: https://github.com/SigNoz/opentelemetry-collector-contrib/pull/2072

### CODE REVIEW

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 1]
```diff
@@ -1556,9 +1558,9 @@ func (r *ClickHouseReader) GetUsage(ctx context.Context, queryParams *model.GetU
 
 	var query string
 	if len(queryParams.ServiceName) != 0 {
-		query = fmt.Sprintf("SELECT toStartOfInterval(timestamp, INTERVAL %d HOUR) as time, count(1) as count FROM %s.%s WHERE serviceName='%s' AND timestamp>='%s' AND timestamp<='%s' GROUP BY time ORDER BY time ASC", queryParams.StepHour, r.traceDB, r.indexTable, queryParams.ServiceName, strconv.FormatInt(queryParams.Start.UnixNano(), 10), strconv.FormatInt(queryParams.End.UnixNano(), 10))
+		query = fmt.Sprintf("SELECT toStartOfInterval(hour, INTERVAL %d HOUR) as time, sum(count) as count FROM %s.%s WHERE serviceName='%s' AND timestamp>='%s' AND timestamp<='%s' GROUP BY time ORDER BY time ASC", queryParams.StepHour, r.traceDB, r.indexTable, queryParams.ServiceName, strconv.FormatInt(queryParams.Start.UnixNano(), 10), strconv.FormatInt(queryParams.End.UnixNano(), 10))
```
Comment (ankitnayan) [inline pkg/query-service/app/clickhouseReader/reader.go:1561]: what is the `hour` field? Shouldn't it be `timestamp`?

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 2]
```diff
@@ -1556,9 +1558,9 @@ func (r *ClickHouseReader) GetUsage(ctx context.Context, queryParams *model.GetU
...
 	} else {
-		query = fmt.Sprintf("SELECT toStartOfInterval(timestamp, INTERVAL %d HOUR) as time, count(1) as count FROM %s.%s WHERE timestamp>='%s' AND timestamp<='%s' GROUP BY time ORDER BY time ASC", queryParams.StepHour, r.traceDB, r.indexTable, strconv.FormatInt(queryParams.Start.UnixNano(), 10), strconv.FormatInt(queryParams.End.UnixNano(), 10))
+		query = fmt.Sprintf("SELECT toStartOfInterval(hour, INTERVAL %d HOUR) as time, sum(count) as count FROM %s.%s WHERE timestamp>='%s' AND timestamp<='%s' GROUP BY time ORDER BY time ASC", queryParams.StepHour, r.traceDB, r.indexTable, strconv.FormatInt(queryParams.Start.UnixNano(), 10), strconv.FormatInt(queryParams.End.UnixNano(), 10))
```
Comment (ankitnayan) [inline pkg/query-service/app/clickhouseReader/reader.go:1563]: please convert this query to prepared statements

**Review by ankitnayan — CHANGES_REQUESTED** [review]

File: pkg/query-service/app/clickhouseReader/reader.go (hunk 1 above)
Comment (makeavish) [reply pkg/query-service/app/clickhouseReader/reader.go:1561]: Haven't pushed the changes as changes were made to otel collector.
Pushing now.

File: pkg/query-service/app/clickhouseReader/reader.go (hunk 2 above)
Comment (makeavish) [reply pkg/query-service/app/clickhouseReader/reader.go:1563]: Sure, updating it.

**Review by srikanthccv — APPROVED** [review]

**Comment by makeavish** [comment]
@ankitnayan Please review and merge.

**Review by ankitnayan — APPROVED** [review]

**Pull request merged**